var/spack/repos/builtin/packages/py-pennylane-lightning/package.py

```diff
@@ -27,7 +28,7 @@ class PyPennylaneLightning(CMakePackage, PythonExtension):
         default=True,
         description="Build with AVX2/AVX512 gate automatic dispatching support",
     )
-    variant("kokkos", default=True, description="Build with Kokkos support")
+    variant("kokkos", default=True, description="Build with Kokkos support", when="@:0.31")
```

Comment:
just to confirm, so from 0.31 onwards all Kokkos dependencies are embedded in the other package, `py-pennylane-lightning-kokkos`, correct?

Reply:
and so, shall we also add a dependency on version at line 79 below, `if "+kokkos" in self.spec:` ?
I know the flag would be ignored, but I would reckon it is neater to make this explicit in the recipe.

Reply:
> just to confirm, so from 0.31 onwards all Kokkos dependencies are embedded in the other package, `py-pennylane-lightning-kokkos`, correct?

No, `py-pennylane-lightning-kokkos` really implements Kokkos kernels (and does not depend on `kokkos-kernels`. Previously, `py-pennylane-lightning` depended on `kokkos-kernels` (and hence `kokkos`) for sparse linear algebra, but it now implements its own routines so that neither is required anymore.